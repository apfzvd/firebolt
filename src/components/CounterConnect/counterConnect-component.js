import React from 'react'
import { Button } from 'grimorio-ui'

export default function Counter({ counter, increment, decrement, reset }) {
  return (
    <div>
      <h1>Componente conectado</h1>
      <div>{counter}</div>
      <Button onClick={increment}>Aumentar valor</Button>
      <Button onClick={decrement}>Diminuir valor</Button>
      <Button onClick={reset}>Zerar</Button>
    </div>
  )
}
