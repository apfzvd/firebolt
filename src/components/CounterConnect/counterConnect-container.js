import { connect } from 'react-redux'
import Counter from './counterConnect-component'
import { increment, decrement, reset } from './counterConnect-action'

export default connect(
  state => ({
    counter: state.totalCounter.counter
  }),
  {
    increment,
    decrement,
    reset
  }
)(Counter)
