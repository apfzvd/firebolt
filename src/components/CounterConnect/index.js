import Counter from './counterConnect-container'

export default Counter
