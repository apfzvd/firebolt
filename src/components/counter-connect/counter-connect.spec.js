import { shallow } from 'enzyme'
import React from 'react'
import CounterConnect from './counter-connect-component'
import reducer, { INITIAL_STATE } from './counter-connect-reducer'
import { increment, decrement, reset } from './counter-connect-actions'

const wrapper = shallow(<CounterConnect />)

describe('Render element', () => {
  it('snapshot counter connect', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
})

describe('Counter reducer', () => {
  it('NONE', () => {
    const state = reducer(undefined, {})
    expect(state).toStrictEqual(INITIAL_STATE)
  })

  it('INCREMENT', () => {
    const state = reducer(INITIAL_STATE, increment())
    expect(state.counter).toStrictEqual(1)
  })

  it('DECREMENT', () => {
    const state = reducer(INITIAL_STATE, decrement())
    expect(state.counter).toStrictEqual(-1)
  })

  it('RESET', () => {
    const state = reducer(INITIAL_STATE, reset())
    expect(state.counter).toStrictEqual(0)
  })

  it('DEFAULT', () => {
    const state = reducer(INITIAL_STATE, {})
    expect(state.counter).toStrictEqual(0)
  })
})
