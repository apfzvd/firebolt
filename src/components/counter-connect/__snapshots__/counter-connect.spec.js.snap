// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render element snapshot counter connect 1`] = `
"<div className={[undefined]}>
  <div>
    <span className={[undefined]}>
      Componente conectado - 
      <strong>
        Valor: 
        0
      </strong>
    </span>
  </div>
  <div>
    <Button onClick={{...}} className={[undefined]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Aumentar valor
    </Button>
    <Button onClick={{...}} className={[undefined]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Diminuir valor
    </Button>
    <Button onClick={{...}} className={[undefined]} disabled={false} color=\\"primary\\" size=\\"medium\\" type=\\"button\\" active={false}>
      Zerar
    </Button>
  </div>
</div>"
`;
